common/workunit/workunit.cpp

```diff
@@ -5060,6 +5114,232 @@ bool CWorkUnitFactory::restoreWorkUnit(const char *base, const char *wuid, bool
+            if (!di->first())
+            {
+                if (optional)
+                    return false;
+                throw MakeStringException(WUERR_InvalidUserInput, "Failed to found %s in %s.", mask, unzipDir.str());
+            }
+
+            fileName.set(unzipDir).append(PATHSEPSTR);
+            di->getName(fileName);
+            return true;
+        }
+        void readWUXMLFileToPTree(Owned<IPTree> &tree, const char *pattern, bool optional)
+        {
+            StringBuffer fileName;
+            if (!findZAPFile(pattern, optional, fileName))
+                return;
+
+            tree.setown(createPTreeFromXMLFile(fileName));
+            if (!tree)
+                throw MakeStringException(WUERR_InvalidUserInput, "Failed to retrieving %s.", pattern);
+
+            Owned<IFile> f = createIFile(fileName);
+            f->remove();
+        }
+        void updateWUProcessLogAttrs(const char *xpath, const char *unzipDirWithIP, bool thorCluster)
+        { 
+            Owned<IPropertyTreeIterator> it = wuTree->getElements(xpath);
+            ForEach (*it)
+            {
+                StringBuffer process, logSpec;
+                IPropertyTree &proc = it->query();
+                proc.getName(process);
+                proc.getProp("@log", logSpec);
+                if (process.isEmpty() || logSpec.isEmpty())
+                    continue;
+
+                //Match the path and the name of the log files from ZAP report.
+                StringBuffer newLogAttr(unzipDirWithIP);
+                newLogAttr.append(process).append('_').append(pathTail(logSpec));
+                proc.setProp("@log", newLogAttr);
+                proc.setPropBool("@singleLog", true);
+
+                if (!thorCluster)
+                    continue;
+
+                StringBuffer slaveLogPattern;
+                slaveLogPattern.set(process).append("_");
+                if (proc.hasProp("@pattern"))
+                {
+                    proc.getProp("@pattern", slaveLogPattern);
+                }
+                else
+                {   //legacy WU
+                    setSlaveLogPatternForLegacyWU(logSpec, slaveLogPattern);
+                }
+                proc.setProp("@pattern", slaveLogPattern);
+
+                if (!proc.hasProp("@max"))
+                    proc.setPropInt("@max", findNumberOfThorSlavesFromSlaveLogs(unzipDirWithIP));
+            }
+        }
+        void setSlaveLogPatternForLegacyWU(const char *logSpec, StringBuffer &slaveLogPattern)
+        {
+            StringBuffer logDate;
+            getThorLogDateFromLogFileName(logSpec, "/thormaster.", logDate);
+
+            slaveLogPattern.append("thorslave.").append(SLAVEIDSTR);
+            if (!logDate.isEmpty()) //Should not happen
+                slaveLogPattern.append(".").append(logDate);
+            slaveLogPattern.append(".log");
+        }
+        unsigned findNumberOfThorSlavesFromSlaveLogs(const char *zapWUFolderPath)
```

Comment:
is this just for legacy wuids?
If so, could do with comment to that effect here.

Reply:
Will add a comment.